lib/utils/autocomplete.js

```diff
@@ -0,0 +1,31 @@
+'use strict';
+
+const homedir = require('os').homedir();
+const readFileSync = require('./fs/readFileSync');
+const fileExistsSync = require('./fs/fileExistsSync');
+
+const tab = require('tabtab')({
+  name: 'serverless',
+});
+
+const autocomplete = () => {
+  const cacheFilePath = `${homedir}/.serverless/autocomplete.json`;
+  if (!fileExistsSync(cacheFilePath)) {
+    return;
+  }
+  const cacheFile = readFileSync(cacheFilePath);
```

Comment:
The cache is not guaranteed to be valid just because it exists. Users could change the plugins listed in `servereless.yml` which would change the outcome of autocomplete. To overcome this I would hash serverless.yml as a cache validation key and store it in the `autocomplete.json`. The autocomplete should then hash `serverless.yml` on every invocation and use that value to compare against the exisitng cache key in `autocomplete.json`. If the keys are different then the cache is invalid and `serverless.init()` should be called to regenerate it.